src/cloud-providers/table.go

```diff
@@ -12,7 +18,57 @@ type CloudProvider interface {
 
 var providerTable map[string]CloudProvider = make(map[string]CloudProvider)
 
+func getSha256sum(providerPath string) (string, error) {
+	file, err := os.Open(providerPath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hash := sha256.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return "", err
+	}
+	sum := hash.Sum(nil)
+	return fmt.Sprintf("%x", sum), nil
+}
+
+// LoadCloudProvider loads cloud provider plugin from the given path
+// "${CLOUD_PROVIDER_PLUGIN_PATH}\${CLOUD_PROVIDER}.so"
+// and call the InitCloud function from the found .so files after the sha256sum is verified
+func LoadCloudProvider(name string) {
```

Comment:
This method will need to be updated to load all the plugins or specific plugins based on configuration?

Reply:
From user point view, they will only use one `CLOUD_PROVIDER` based on configmap in production env, so I assume that we only load the specific `CLOUD_PROVIDER` plugin is enough here.

Reply:
Like with built-in providers, there could be multiple plugins loaded but only one used by changing the peer-pods-cm. But I'm ok if we want to only support a single external plugin given CAA can work with only one.

Reply:
We can put multiple external plugins .so files to the ${CLOUD_PROVIDER_PLUGIN_PATH} folder on worker node, but we do NOT need load them all, we only need load the one define in peer-pods-cm, if we want to use other `CLOUD_PROVIDER` we can only changing the setting in peer-pods-cm: `CLOUD_PROVIDER` and `CLOUD_PROVIDER_EXTERNAL_PLUGIN_HASH`